src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -737,11 +990,18 @@ internal static bool IsZeroElementTupleType(TypeSymbol type)
                 deconstructionSubpatterns = patternsBuilder.ToImmutableAndFree();
             }
 
-            ImmutableArray<BoundSubpattern> properties = default;
-            if (node.PropertyPatternClause != null)
-            {
-                properties = BindPropertyPatternClause(node.PropertyPatternClause, declType, inputValEscape, permitDesignations, diagnostics, ref hasErrors);
-            }
+            ImmutableArray<BoundSubpattern> properties = node.PropertyPatternClause.IsKind(SyntaxKind.PropertyPatternClause)
+                ? BindPropertyPatternClause(node.PropertyPatternClause, declType, inputValEscape, permitDesignations, diagnostics, ref hasErrors)
+                : default;
+
+            PropertySymbol? lengthProperty = null;
+            BoundPattern? lengthPattern = node.LengthPatternClause is not null
+                ? BindLengthPatternClause(node.LengthPatternClause, declType, permitDesignations, out lengthProperty, ref hasErrors, diagnostics)
```

Comment:
Good catch using `declType` instad of `inputType`. (I see this is covered by `ListPattern_NarrowedTypes`)